src/client/auth/sasl.rs

```diff
@@ -67,9 +67,11 @@ impl SaslContinue {
 
 /// Validates that a `saslStart` or `saslContinue` command response is successful.
 fn validate_command_success(auth_mechanism: &str, response: &Document) -> Result<()> {
-    let ok = response
-        .get("ok")
-        .ok_or_else(|| Error::invalid_authentication_response(auth_mechanism))?;
+    let ok = match response.get("ok") {
```

Comment:
The server doesn't send back `"ok": 1` in the speculative auth response document, so the simplest solution is to just treat a lack of any `"ok"` field as the same as `"ok": 1`. I left in the check that if the field is present, it must be `1` (or `1.0`).